openpype/modules/deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -190,17 +189,23 @@ def inject_openpype_environment(deadlinePlugin):
             raise RuntimeError(msg)
 
         if not os.environ.get("OPENPYPE_MONGO"):
-            print(">>> Missing OPENPYPE_MONGO env var, process won't work")
+            if not job.GetJobEnvironmentKeyValue('OPENPYPE_MONGO'):
+                print(">>> Missing OPENPYPE_MONGO env var, process won't work")
 
         os.environ["AVALON_TIMEOUT"] = "5000"
 
         args_str = subprocess.list2cmdline(args)
         print(">>> Executing: {} {}".format(exe, args_str))
-        process = ProcessUtils.SpawnProcess(
-            exe, args_str, os.path.dirname(exe)
+
+        current_env = {**os.environ}
+        current_env["OPENPYPE_MONGO"] = job.GetJobEnvironmentKeyValue(
+            'OPENPYPE_MONGO'
         )
-        ProcessUtils.WaitForExit(process, -1)
-        if process.ExitCode != 0:
+
+        process = subprocess.Popen([exe] + args, env=current_env)
```

Comment:
Will deadline monitor show output from the process when `subprocess.Popen` is used?

Reply:
I think so. Maybe the log I added answers the question.